index.js

```diff
@@ -39,12 +39,13 @@ const getStreamContents = async (stream, {convertChunk, getSize, getContents}, {
 		for await (const chunk of stream) {
 			const chunkType = getChunkType(chunk);
 			const convertedChunk = convertChunk[chunkType](chunk, textDecoder);
-			chunks.push(convertedChunk);
 			length += getSize(convertedChunk);
```

Comment:
Wouldn't the `length` be wrong here for the `bufferedData` on the error? Since when there is a `MaxBufferError` we skip the new chunks, but we still included the chunk size in the `length` sum.

Reply:
You're right!
Fixed in 35327d0646e16e8374addf1066ef9248ad48aebe

The test was not catching this because it was using `getStream()`, which does not use `length` to concatenate all chunks. I switched it to `getStreamAsBuffer()`, and the test was now failing. It is passing now.